homeassistant/components/esphome/update.py

```diff
@@ -133,6 +145,19 @@ def _static_info_updated(infos: list[EntityInfo]) -> None:
             )
         )
 
+        @callback
+        def _on_device_update() -> None:
+            """Handle update of device state, like availability."""
+            self.async_write_ha_state()
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"esphome_{self._entry_id}_on_device_update",
```

Comment:
For `static_info_updated` I moved the signal creation to the EntryData, maybe we can do it here too? `self._entry_data.signal_device_updated`

Reply:
Done 👍